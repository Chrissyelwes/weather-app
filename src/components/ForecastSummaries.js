import React from "react";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";

function ForecastSummaries({ forecasts }) {
  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => (
        <ForecastSummary
          key={forecast.date}
          date={forecast.date}
          description={forecast.description}
          icon={forecast.icon}
          temperature={forecast.temperature}
        />
      ))}
    </div>
  );
}

export default ForecastSummaries;

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
    })
  ).isRequired,
};
