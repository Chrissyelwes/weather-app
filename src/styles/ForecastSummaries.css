.forecast-summaries {
    display: flex;
    flex-direction: row;
    justify-content: center;
    gap: 50px;
}

@media only screen and (max-width: 600px) {
    .forecast-summaries {
        display: flex;
        flex-direction: column;
        justify-content: center;
        gap: 50px;
    }
}