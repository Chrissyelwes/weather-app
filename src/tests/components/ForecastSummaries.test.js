import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1111111,
        description: "Stub description 1",
        icon: "800",
        temperature: {
          max: 22,
          min: 12,
        },
      },
      {
        date: 3333333,
        description: "Stub description2",
        icon: "211",
        temperature: {
          max: 8,
          min: 13,
        },
      },
    ],
    onForecastSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries
        forecasts={validProps.forecasts}
        onForecastSelect={validProps.onForecastSelect}
      />
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
