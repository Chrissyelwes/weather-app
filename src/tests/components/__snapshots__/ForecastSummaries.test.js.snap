// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        800
      </div>
      <div
        class="forecast-summary_temperature"
      >
        22°c
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description 1
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary_date"
      >
        Thu Jan 01 1970
      </div>
      <div
        class="forecast-summary_icon"
        data-testid="forecast-icon"
      >
        211
      </div>
      <div
        class="forecast-summary_temperature"
      >
        8°c
      </div>
      <div
        class="forecast-summary_description"
      >
        Stub description2
      </div>
      <button
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
